### PR TITLE
Adds RFC for cabbage roadmap for start of 2022

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Closes https://github.com/giantswarm/giantswarm/issues/20883
Towards https://github.com/giantswarm/giantswarm/issues/18676

This RFC describes the roadmap for Team Cabbage for the first part of 2022.

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
Starting with reviews from cabbage, then i'll open it up to sig product.

**Comment by JosephSalisbury** [comment]
yolo, let's go sig product

File: cabbage-roadmap-2022-1/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+Before moving to the more concrete opportunities, it's useful to cover more long-term patterns and use cases we're thinking about. These are items that are not yet ready for us to tackle, but are going to form the basis for our product discovery over the new few months, and will likely form part of our roadmap in the future.
+
+We also need to make sure that we take the current strategic move to Cluster API into account.
+
+### Gateway API
+
+See the [upstream project](https://gateway-api.sigs.k8s.io/).
+
+Gateway API can be described as a possible future replacement for the current [Ingress API object](https://kubernetes.io/docs/concepts/services-networking/ingress/), aiming to provide a more well-rounded API for describing service networking in Kubernetes.
```
Comment (puja108) [inline cabbage-roadmap-2022-1/README.md:35]: I think we could mention here already that one of the main use cases is supporting better multi-tenancy and separation of concerns within the use case of exposing services.

File: cabbage-roadmap-2022-1/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
+# Team Cabbage Roadmap 2022
+
+This RFC describes the roadmap for Team Cabbage for the first part of 2022.
```
[the remark this answers is not in the document]
Comment (JosephSalisbury) [reply cabbage-roadmap-2022-1/README.md:3]: Yeah, I don't know if it's just for Q2, or Q2 or Q3, or if we'll re-evaluate everything next month. So, staying vague intentionally :D

File: cabbage-roadmap-2022-1/README.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+Finally, we continued improving our support for shipping Kong as a Managed App, and shipped the AWS LB Controller as a Managed App. 
+
+## Jobs To Be Done
+
+The Jobs To Be Done for Team Cabbage can be summarised as follows:
+
+- Exposing services running on a Kubernetes cluster
+- Routing, observing, and securing traffic into, inside, and out of Kubernetes clusters
+- Managing API microservices on Kubernetes clusters
```
[the remark this answers is not in the document]
Comment (JosephSalisbury) [reply cabbage-roadmap-2022-1/README.md:19]: I didn't intend for any specific intention w/ 'microservices' - i'm thinking APIs in a very general sense

**Review by puja108 — COMMENTED** [review]
minor nitpicks, otherwise LGTM, nice to see old school roadmap dos making a comeback

**Review by ubergesundheit — APPROVED** [review]

File: cabbage-roadmap-2022-1/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (JosephSalisbury) [reply cabbage-roadmap-2022-1/README.md:19]: moved to 'API services', I think that's generic enough

File: cabbage-roadmap-2022-1/README.md (hunk 1 above)
Comment (JosephSalisbury) [reply cabbage-roadmap-2022-1/README.md:35]: Added a bit lower

**Comment by JosephSalisbury** [comment]
@puja108 please take another look and hit me w/ an approve if you like

@weatherhog i'd also like comments / an approve from you too before merging

**Review by puja108 — APPROVED** [review]

**Review by weatherhog — APPROVED** [review]

File: cabbage-roadmap-2022-1/README.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+We have some customers with extremely bursty traffic patterns, meaning they need to pre-scale their clusters before high load events. We would likely not be able to enable them to scale on demand, even with the ability to scale on custom metrics.
+
+The effort for this is somewhat middling, but so is the customer impact.
+
+### External monitoring of Ingress Controllers
+
+We currently monitor Ingress Controllers from the monitoring systems deployed on the Management Cluster. This can be problematic in situations where the Ingress Controllers are seen as available to our monitoring systems, but are not available to the Internet as a whole, such as with some forms of network failure.
+
+We could provide an monitoring service that tests whether Ingress Controllers are accessible from the Internet.
```
[the remark this answers is not in the document]
Comment (JosephSalisbury) [reply cabbage-roadmap-2022-1/README.md:103]: I think I'd leave that to be decided somehow - it feels like part of the design of the solution.

**Pull request merged**